### ISSUE TITLE
Not able to utilise Gpu properly

### ISSUE THREAD

**Issue by akashAD98** [ultralytics/ultralytics#1826]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

system ubantu 20.04
RTX 3090 24 GB,16 cores

I'm training my custom model,& I have rtx 3090 GPU 24 GPU ram memory & 16 cores, 

when i put batch 32= its using 8/24 memory & my CPU usage is shown very high like few of cores are reaching 100  % & my system is automatically hanging & shutdown. 

<img width="767" alt="image" src="https://user-images.githubusercontent.com/62583018/229988994-af13a406-ccc8-4ca5-bad0-54f2829d1371.png">

### Additional

so i put batchsize=16 & started trainig ,but its ionly using 4 gb/24 gb ,so im not able to utilise propely gpu 

<img width="552" alt="image" src="https://user-images.githubusercontent.com/62583018/229989340-6deadd32-5efc-4e1e-bf5d-740584561380.png">

**Comment by glenn-jocher** [comment]
@akashAD98 based on the information you provided, it seems like your GPU memory usage is not being fully utilized when using a batch size of 16, despite the large amount of memory available on your RTX 3090 GPU. It is possible that your system is hanging and shutting down due to high CPU usage when using a batch size of 32, which could be causing your CPU cores to reach 100%. It may be useful to track your GPU and CPU usage during training to better understand how they are being utilized. Additionally, you may want to try adjusting your training parameters, such as adjusting learning rate or adding weight decay, to help improve performance and make better use of your GPU resources. Ultimately, identifying and addressing the root cause of your system issues will require further investigation and possibly seeking help from a system administrator or technical support.

**Comment by AyushExel** [comment]
To add to what Glenn said, it is possible that your CPU or HDD is causing a bottleneck while reading the dataset. Can you try to cache the dataset and see if it improves anything?

**Comment by akashAD98** [comment]
@AyushExel im using cache=True now,hopeing it will work now

**Comment by AyushExel** [comment]
@akashAD98 okay. If you run of out memory, maybe you can use `cache=disk` to cache on HDD. Btw do you have SSD or HDD?

**Comment by akashAD98** [comment]
@AyushExel till  now no  error, I have ssd in my system

**Comment by glenn-jocher** [comment]
@akashAD98 great to hear that using `cache=True` has resolved your issue. Caching allows you to store data in memory instead of repeatedly reading from disk during training, which can improve performance. If you continue to have issues with training performance and GPU utilization, adjusting other training parameters such as the learning rate or data augmentation can also help. It may also be useful to monitor your GPU and CPU usage during training to better understand how your system resources are being utilized.
